REF: FloatButtons component

limpbrains opened the pull request:
Problem: 
There are 3 very alike components: 
- Scan button on main screen
- Send/recieve buttons on wallet/list screen
- Undo/save on provide entropy screen
All of them looks the same
But not all of them working correctly with different languages.

Solution:
Create universal component with nice UI that supports any number of buttons. Also, they should work for any language.

Before:
<img width="385" alt="image" src="https://user-images.githubusercontent.com/155891/92409285-c13a8a80-f148-11ea-9be7-92432c1894aa.png">
<img width="387" alt="image" src="https://user-images.githubusercontent.com/155891/92409307-d3b4c400-f148-11ea-9a90-e465300e3774.png">

After:
<img width="383" alt="image" src="https://user-images.githubusercontent.com/155891/92409351-f515b000-f148-11ea-88d9-2c3154534fc8.png">
<img width="381" alt="image" src="https://user-images.githubusercontent.com/155891/92409375-11b1e800-f149-11ea-8f9f-79f8a4157e12.png">

It also works with 3 buttons:
<img width="394" alt="image" src="https://user-images.githubusercontent.com/155891/92409424-3f972c80-f149-11ea-8adb-cf0ba7c5c3da.png">

Two buttons with different content width will still have the same size
<img width="386" alt="image" src="https://user-images.githubusercontent.com/155891/92409466-648b9f80-f149-11ea-930e-fc2c457433f0.png">

To make it happen it firsts render buttons outside the screen to calculate their sizes. Then calculate container width render them on screen

API:
```js

import { FContainer, FButton } from '../../components/FloatButtons';
<FContainer>
  <FButton
    onPress={() => console.log('down')}
    icon={<Icon name="arrow-down"/>}
    text={loc.receive.down}
  />
  <FButton
    onPress={() => console.log('up')}
    icon={<Icon name="arrow-up"/>}
    text={loc.receive.up}
  />
</FContainer>
```

limpbrains commented:
Little problem:
Before there was one place where `size` for `<Icon` was calculated. In `BlueComponents.js`
Now because Icon provided as a paratemeter to `FButton` there are 3:
FloatButtons, screen/provideEntropy, screen/wallet/transactions
Can we move such calculations to `BlueDefaultTheme` ? @marcosrdz what do you think?

GladosBlueWallet commented:
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/de2dvfwrchuwjnxdacpbehpenr

[download apk](http://filestorage.bluewallet.io:1488/1599507821.apk)

marcosrdz commented:
Conflicts

limpbrains commented:
@marcosrdz fixed

GladosBlueWallet commented:
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/d3fqkrxdnxxru67f2vzfp1m4er

[download apk](http://filestorage.bluewallet.io:1488/1599677386.apk)

marcosrdz commented:
@limpbrains  button doesnt highlight when tapping, only text does.

Overtorment commented:
indeed button doesnt highlight tap. also, text is visibly cropped on appetize:

![image](https://user-images.githubusercontent.com/1913337/93667726-f798d380-fa7f-11ea-8323-6e684fe67542.png)

Overtorment commented on screen/wallets/transactions.js:
```diff
@@ -630,6 +666,12 @@ const WalletTransactions = () => {
     }
   };
+  console.info('RECIEVE', wallet.allowReceive());
+
```
debug prints

Overtorment reviewed (changes requested):
mostly lgtm

limpbrains commented on components/FloatButtons.js:
```diff
@@ -0,0 +1,115 @@
+  }
+
+  return (
+    <TouchableOpacity style={[bStyles.root, style]} {...props}>
```
Let's try when we will have RN 0.63, we are on 0.62 currently

GladosBlueWallet commented:
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/rdq1n2m68m7yj6cuhg996cx1kw

[download apk](http://filestorage.bluewallet.io:1488/1600600073.apk)

limpbrains commented:
@marcosrdz @Overtorment all is fixed, please review again

Overtorment reviewed (approved):
lgtm

ncoelho reviewed (changes requested):
- There a crash when opening watch-only
- Scan size seems very small in english
- The buttons are loading after the screen loads, is this on purpose?

GladosBlueWallet commented:
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/7ubuxu8hjxyu1xuugepc27xhg4

[download apk](http://filestorage.bluewallet.io:1488/1600683143.apk)

limpbrains commented:
>     * There a crash when opening watch-only
fixed
>     * Scan button size seems very small in english, are the paddings the same as before?
fixed for Scan button
>     * The buttons are loading after the screen loads, is this on purpose?
Not on purpose. This component renders in 2 steps. On the wallet screen there is list of transaction component it blocks JS for a second, this prevents FloatButtons to appear.
Do you think this is critical ?

GladosBlueWallet commented:
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/9005ykk3kuv1rb0txc5k399z0w

[download apk](http://filestorage.bluewallet.io:1488/1600684348.apk)

ncoelho commented:
> Do you think this is critical ?

Not at all, just asking because it is noticeable on slower devices. Making it look slower is never good.

GladosBlueWallet commented:
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/13nkp3b8t7r07nxbnqq5wc4vjm

[download apk](http://filestorage.bluewallet.io:1488/1600947255.apk)

ncoelho commented:
Check padding on two buttons case. Should be:

_button1_ + _button2_

<img width="244" alt="Screenshot 2020-09-30 at 17 21 34" src="https://user-images.githubusercontent.com/1238429/94705631-8f7ca400-0341-11eb-8763-c5935d9d2318.png">

limpbrains commented:
Because buttons have different content length I can't use grid. 

Master branch. Here content of Send button is center aligned:
<img width="417" alt="изображение" src="https://user-images.githubusercontent.com/155891/94910088-df2bae80-04ac-11eb-8e6a-2aa53f857280.png">

I've increased button paddings in the middle, so it is mo accurate now. My branch:
<img width="415" alt="изображение" src="https://user-images.githubusercontent.com/155891/94910910-0c2c9100-04ae-11eb-8ad2-88b95dd63796.png">

Very long send button example:
<img width="414" alt="изображение" src="https://user-images.githubusercontent.com/155891/94910351-4ba6ad80-04ad-11eb-83a8-f6db4f2f155b.png">

ncoelho commented:
Looks good, thanks for checking all the details.

GladosBlueWallet commented:
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/xdwtfun8jg6vn6f5dnpntbg75m

[download apk](http://filestorage.bluewallet.io:1488/1601633630.apk)

Overtorment reviewed (changes requested):
pls fix conflicts

GladosBlueWallet commented:
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/52qkbbjhv917cdyueydzjqnxcm

[download apk](http://filestorage.bluewallet.io:1488/1602144486.apk)

limpbrains commented:
> pls fix conflicts

fixed
